plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergConfig.java

```diff
@@ -55,6 +55,11 @@ public class IcebergConfig
     public static final String EXPIRE_SNAPSHOTS_MIN_RETENTION = "iceberg.expire_snapshots.min-retention";
     public static final String REMOVE_ORPHAN_FILES_MIN_RETENTION = "iceberg.remove_orphan_files.min-retention";
 
+    public enum BucketExecutionMode
+    {
+        NEVER, AUTO, ALWAYS
```

Comment:
Connector cannot enforce engine to user partitioning unless it's fixed bucket to node mapping.
Decision to use use partitioning or not should be governed by `DetermineTableScanNodePartitioning`.
For Iceberg partitions `DetermineTableScanNodePartitioning` doesn't currently work because we lack estimation of number of partitions (See the discussion here https://github.com/trinodb/trino/issues/19062). There are also more factors at play here which could affect `DetermineTableScanNodePartitioning` decision:
* skewness of partitions
* number of partitions in relation to size of query
Connector shouldn't have to deal with these

Reply:
This was a WIP and this part got rewritten in the final code